geocodingsearchv7/batchgeocoder_test.go

```diff
@@ -0,0 +1,108 @@
+package geocodingsearchv7_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"testing"
+
+	"go.einride.tech/here/geocodingsearchv7"
+	"gotest.tools/v3/assert"
+)
+
+type BatchGeocodingMock struct {
+	responseStatus int
+	responseBody   geocodingsearchv7.BatchGeocoderResponse
+}
+
+func (c *BatchGeocodingMock) Do(req *http.Request) (*http.Response, error) {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	b, err := json.Marshal(c.responseBody)
+	if err != nil {
+		return nil, err
+	}
+	r := bytes.NewReader(b)
+	return &http.Response{
+		StatusCode:    c.responseStatus,
+		Header:        headers,
+		Body:          io.NopCloser(r),
+		ContentLength: int64(len(b)),
+	}, nil
+}
+
+func TestBatchGeocodingService_BatchGeocode(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
```

Comment:
Could it be dangerous to share the same context between the different tests?
Could it be better to let each test define it's own context?

This comment applies to all tests in this file.